docs/source/developers/python.rst

```diff
@@ -422,16 +447,19 @@ Now, we can build pyarrow:
 
 .. code-block:: shell
 
-   cd python
+   pushd arrow\python
    set PYARROW_WITH_GANDIVA=1
    set PYARROW_WITH_PARQUET=1
    python setup.py build_ext --inplace
+   popd
 
 Then run the unit tests with:
 
 .. code-block:: shell
 
+   pushd arrow\python
    py.test pyarrow -v
+   popd
```

Comment:
At line 474, I cannot execute `python-test.exe` or `ctest` at `arrow\python`. I may miss something to build `python-test.exe`

```
(pyarrow-dev) C:\arrow\arrow\python>ctest
*********************************
No test configuration file found!
*********************************
Usage

  ctest [options]


(pyarrow-dev) C:\arrow\arrow\python>python-test
'python-test' is not recognized as an internal or external command,
operable program or batch file.

(pyarrow-dev) C:\arrow\arrow\python>dir python-test*
 Volume in drive C is OS
 Volume Serial Number is 7884-B5DB

 Directory of C:\arrow\arrow\python

File Not Found
```

Reply:
I had not tried this yet. Can confirm your observations.

Could this be an obsolete part of the instructions? In the linux instructions neither `python-test` nor `ctest` seem to make an appearance.

Reply:
You need to pass `-DARROW_BUILD_TESTS=ON` to build `arrow-python-test.exe`, the unit tests are now toggled off by default. I don't think most developers need to build this executable but this should go in a separate section for completeness

Reply:
@wesm Thanks. `ctest` runs now but it must be run from the `cpp/build` directory.

Also, `arrow-python-test` fails:
```
      Start 58: arrow-python-test
58/83 Test #58: arrow-python-test .........................Exit code 0xc0000409
***Exception:   0.51 sec
```

I can find a `arrow-python-test.exe` in `cpp/build/release/Release`. Unfortunately running it directly is not working either.

```
(pyarrow-dev) Z:\dev\arrow\cpp\build\release\Release>arrow-python-test.exe
Fatal Python error: initfsencoding: unable to load the file system codec
ModuleNotFoundError: No module named 'encodings'

Current thread 0x00002390 (most recent call first):
```

Reply:
@wesm Apologies. `ctest` and `arrow-python-test.exe` do run ok.

I forgot the `set PYTHONHOME=%CONDA_PREFIX%`...

Reply:
Thank you. This change also works well for my environment.

Reply:
It's not necessary for most developers to run this unit tests, so we should put this off in a separate Advanced section

Reply:
I see this is already done